### DISCUSSION TITLE
[CodeStyle] Use `var` with discretion.

### DISCUSSION THREAD

**Discussion by lyndsiWilliams** [Studio-Lovelies/GG-JointJustice-Unity#162] [Ideas]
### Formerly: `Code styles revamp`

---

> Use `var` with discretion. If `var` is found to be confusing during a pr, comment on it to have it changed

---

[Our current code styles doc](https://github.com/Studio-Lovelies/GG-JointJustice-Unity/wiki/Code-Style-Guideline) seems a little outdated. It's been a while since we've created/revisited it, and there are a couple of rules that could be adjusted. For instance, "Use var with discretion." as a variable rule. We're a little looser on using `var` now, so I think that rule can be changed/removed. Are there any other rules we should adjust/remove/add?

**Comment by ViMaSter** [comment]
Thanks for opening up this thread! Definitely got some opinions. : D
I hope you're also okay with me making threads for each rule. That way simple changes can easily be closed up and done and discussions can stay focussed to one rule per thread.

> We're a little looser on using `var` now, so I think that rule can be changed/removed.

+1 from me on that. I personally prefer using `var`, that said, it's purely syntactic sugar. I'm totally fine with just removing the rule altogether instead of adding another that recommends using `var`.

As an aside: Having an example for each rule would be good too. JetBrains' tools do this when displaying formatter changes and I personally find that to be super useful.

https://user-images.githubusercontent.com/1689033/150949750-9784cdd7-0c77-40c3-9586-d4870984b91c.mp4

And I think we're safe to delete all the old wiki pages, [now that the docs live here](https://github.com/Studio-Lovelies/GG-JointJustice-Unity/tree/develop/docs)?

**Reply by PhoebeMitchell** [reply]
>  I think we're safe to delete all the old wiki pages

+1, but let Anasthera know where to find the new docs

**Comment by PhoebeMitchell** [comment]
+1 using var. I've come around to it.
